### PR TITLE
Fix ProxyDAO listen support

### PR DESCRIPTION
**Pull request by kirkeaton**

### CODE REVIEW

File: src/foam/dao/MDAO.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ public FObject put_(X x, FObject obj) {
       }
 
       state_ = index_.put(state_, obj);
+      onPut(obj);
```
Comment (adamvy) [inline src/foam/dao/MDAO.java]: Can we move this outside the writeLock?  Don't need to block other writers while listeners are firing.

File: src/foam/dao/MDAO.java [hunk 2]
```diff
@@ -95,6 +96,7 @@ public FObject remove_(X x, FObject obj) {
 
       if ( found != null ) {
         state_ = index_.remove(state_, found);
+        onRemove(found);
```
Comment (adamvy) [inline src/foam/dao/MDAO.java]: ditto

File: src/foam/dao/MDAO.java (hunk 1 above)
Comment (kirkeaton) [reply src/foam/dao/MDAO.java]: That would require moving the return statement outside of the lock as well. I'm not too dialled into the functionality of the MDAO so I don't know if moving the return values outside of the lock will have an unintended side effect

Comment (kgrgreer) [reply src/foam/dao/MDAO.java]: onPut() and return could both be moved outside the lock.

Comment (kirkeaton) [reply src/foam/dao/MDAO.java]: Done

File: src/foam/dao/MDAO.java (hunk 2 above)
Comment (kirkeaton) [reply src/foam/dao/MDAO.java]: Done

**Pull request merged**